### PR TITLE
dev-cpp/yaml-cpp: enable tests, take package

### PR DESCRIPTION
**Pull request by DarthGandalf**
Package-Manager: Portage-3.0.18, Repoman-3.0.2
Signed-off-by: Alexey Sokolov <alexey+gentoo@asokolov.org>

```diff
--- yaml-cpp-0.6.3-r3.ebuild    2021-06-14 00:27:19.386823736 +0100
+++ yaml-cpp-0.6.3-r4.ebuild    2021-06-14 00:34:42.720628784 +0100
@@ -1,4 +1,4 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
@@ -8,31 +8,31 @@ inherit cmake-multilib
 
 DESCRIPTION="YAML parser and emitter in C++"
 HOMEPAGE="https://github.com/jbeder/yaml-cpp"
-SRC_URI="https://github.com/jbeder/${PN}/archive/${P}.tar.gz"
+SRC_URI="https://github.com/jbeder/yaml-cpp/archive/${P}.tar.gz"
 
 LICENSE="MIT"
 SLOT="0/0.6"
-KEYWORDS="amd64 ~arm arm64 ~hppa ppc ppc64 sparc x86 ~amd64-linux ~x86-linux"
+KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux"
 IUSE="test"
 
-# test breaks build
-#RESTRICT="!test? ( test )"
-RESTRICT+="test"
+RESTRICT="!test? ( test )"
 
-DEPEND="test? ( dev-cpp/gtest )"
+DEPEND="test? ( dev-cpp/gtest[${MULTILIB_USEDEP}] )"
 
-S="${WORKDIR}/${PN}-${P}"
+S="${WORKDIR}/yaml-cpp-${P}"
 
 PATCHES=(
        "${FILESDIR}/${P}-abi-breakage.patch"
        "${FILESDIR}/${P}-CVE-2017-11692.patch"
        "${FILESDIR}/${P}-fix-overflows.patch"
+       "${FILESDIR}/${P}-gtest.patch"
 )
 
 src_prepare() {
        sed -i \
                -e 's:INCLUDE_INSTALL_ROOT_DIR:INCLUDE_INSTALL_DIR:g' \
                yaml-cpp.pc.cmake || die
+       rm -r test/gtest-* || die
 
        cmake_src_prepare
 }
```

### CODE REVIEW

File: dev-cpp/yaml-cpp/yaml-cpp-0.6.3-r4.ebuild [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		-e 's:INCLUDE_INSTALL_ROOT_DIR:INCLUDE_INSTALL_DIR:g' \
+		yaml-cpp.pc.cmake || die
+	rm -r test/gtest-* || die
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DBUILD_SHARED_LIBS=ON
```
Comment (SoapGentoo) [inline dev-cpp/yaml-cpp/yaml-cpp-0.6.3-r4.ebuild:42]: try removing this (default as of EAPI 7)

File: dev-cpp/yaml-cpp/yaml-cpp-0.6.3-r4.ebuild [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+LICENSE="MIT"
+SLOT="0/0.6"
+KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="test"
+
+RESTRICT="!test? ( test )"
+
+DEPEND="test? ( dev-cpp/gtest[${MULTILIB_USEDEP}] )"
+
+S="${WORKDIR}/yaml-cpp-${P}"
```
Comment (SoapGentoo) [inline dev-cpp/yaml-cpp/yaml-cpp-0.6.3-r4.ebuild:22]: put this after SRC_URI

File: dev-cpp/yaml-cpp/yaml-cpp-0.6.3-r4.ebuild [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+DESCRIPTION="YAML parser and emitter in C++"
+HOMEPAGE="https://github.com/jbeder/yaml-cpp"
+SRC_URI="https://github.com/jbeder/yaml-cpp/archive/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0/0.6"
+KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="test"
+
```
Comment (SoapGentoo) [inline dev-cpp/yaml-cpp/yaml-cpp-0.6.3-r4.ebuild:17]: remvoe this empty line

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

**Comment by DarthGandalf** [comment]
Done

**Pull request closed without merging**